### DISCUSSION TITLE
How to add java error log count alerting rule for 9.4.0

### DISCUSSION THREAD

**Discussion by liuvlun** [apache/skywalking#12382] [General]
I implemented the lal and mal and the log metric also can be displayed in widget, but when i add the error count alerting rule, it always failed and cant trigger the condition. 
So please see the following detail and help to check which place is not correct. thank you.
The lal file as below  and it works:
`rules:
  - name: log_count
    layer: GENERAL
    dsl: |
      filter {
        text {
          abortOnFailure true
          regexp "(?<timestamp>\\d{4}-\\d{2}-\\d{2} \\d{2}:\\d{2}:\\d{2}.\\d{3}) \\[TID:(?<tid>.+?)] \\[(?<thread>.+?)] (?<level>\\w{4,})"
        }
        extractor {
          metrics {
            timestamp log.timestamp as Long
            labels level: parsed.level, service: log.service, instance: log.serviceInstance
            name "log_count"
            value 1
          }
        }
        sink {
          if (parsed.level == "INFO") {
            enforcer { }
          }
          if (parsed.level == "ERROR") {
            enforcer { }
          }
          if (parsed.level == "WARN") {
            enforcer { }
          }
        }
      }`

The mal file as below and it works
`expSuffix: instance(['service'], ['instance'], Layer.GENERAL)
metricPrefix: log
metricsRules:
  - name: count_info
    exp: log_count.tagEqual('level', 'INFO').sum(['service', 'instance']).downsampling(SUM)
  - name: count_warn
    exp: log_count.tagEqual('level', 'WARN').sum(['service', 'instance']).downsampling(SUM)
  - name: count_error
    exp: log_count.tagEqual('level', 'ERROR').sum(['service', 'instance']).downsampling(SUM)
  - name: count_error_rate
    exp: log_count.tagEqual('level', 'ERROR').sum(['service', 'instance']).downsampling(SUM).increase('PT1M')`

The alerting rule segment as below but it not works:
`  log_count_error_rule:
    metrics-name: log_count_error
    #expression: sum(log_count_error)
    op: ">"
    threshold: 1
    #period: 1
    tags:
      level: WARNING
    #count: 2
    #silence-period: 5
    message: The error logs are more than 5 in latest ten minutes, please have a look `

The metric graph in service instance as below:
![image](https://github.com/apache/skywalking/assets/13976118/b881e7df-0fe6-4f7d-8112-3ccced77b954)
Please help to check what`s the problem, thank you again.

**Comment by wu-sheng** [comment]
9.4 is too old, some old versions don't support MAL metrics for alerting.

I believe the 10.0.1 works for that, and the alerting kernel has been upgraded to use MQE.

**Reply by liuvlun** [reply]
Yeah, i also agree with you, but when i use the 10,0.1 and met this problem, as this disscussion description: [event table not found](https://github.com/apache/skywalking/discussions/12381). please help to check.

**Reply by wu-sheng** [reply]
You can't use the same database of 9.4 to run 10.0 OAP. That is impossible to work.

**Reply by liuvlun** [reply]
I tried mysql 8.4, h2 and banyandb 0.6.1 database, it caused the same error exception. So i dont know how to check the reason.

**Reply by wu-sheng** [reply]
We run all of those, all work.
Again, you can't share the database including old data/index/table but with latest OAP.

The demo is running on 10.0.1 and BanyanDb 0.6.1

**Reply by liuvlun** [reply]
I run the 10.0.1 in a new database, not share the old database.  I am very confused for this version.
I download the 10.0.1 version and run it directly. 
For now, i dont know how to find the reason, could you give a direction how to follow up this problem.

**Reply by wu-sheng** [reply]
Not just a version, SkyWalking runs every commits in a real k8s and make sure metrics, logs, and traces work.

So, I don't have a clue.

**Reply by liuvlun** [reply]
The same version in my window OS is ok, do you think if it is the permission issue of centos?

**Reply by wu-sheng** [reply]
I don't use CentOS. Sorry.

**Reply by liuvlun** [reply]
I will run this version in unbuntu and check it.

**Reply by liuvlun** [reply]
I run V10.0.0 + BanyanDB in centos normal, but found the BanyanDB database run not stable and it stop after running 3hours.
Three questions:
1. How to run BanyanDB in daemon?
2. Why BanyanDB log dont have the timestamp in the log?
3. Why are logs sorted in positive order and how to change the log sort to reverse order

**Reply by wu-sheng** [reply]
BanyanDB is new, you could run elasticsearch if you don't want to dig deeper about codes.

Using BanyanDB at current stage, we recommend you having code reading capabilities.
